tests/server/test_compilerservice.py

```diff
@@ -119,6 +120,54 @@ def write_main(self, main: str, environment: Optional[data.Environment] = None)
         subprocess.check_output(["git", "add", "main.cf"], cwd=self.src_dir)
         subprocess.check_output(["git", "commit", "-m", "write main.cf", "--allow-empty"], cwd=self.src_dir)
 
+    def write_plugin(self, plugin_name: str, plugin_code: str, modules_dir):
```

Comment:
Perhaps we can make this a bit more generic so we can more easily extend it in the future if we need it?
```suggestion
    def write_module(self, module_name: str, *, plugin_code: str, template_dir: str) -> None:
```